Fixes for DSU 2.1

khappone opened the pull request:
DSU 2.1 seems to fetch some index catalogs outside of yum. If your outgoing traffic goes via proxy, this failed. This is now fixed. Also added an optional (default on) alias to manual dsu runs so it automatically gets the proxy.

Also removed the "creates" from the installation of the DSU repo. This is because we may need to run the tool again to install new gpg keys.

khappone committed:
Fixes for DSU 2.1

DSU 2.1 seems to fetch some index catalogs outside of yum. If your
outgoing traffic goes via proxy, this failed. This is now fixed. Also
added an optional (default on) alias to manual dsu runs so it
automatically gets the proxy.

Also removed teh "creates" from teh installation of th DSU repo. This is
because we may need to run the tool again to install new gpg keys.

Klaas- commented:
you can also fetch everything through dnf via source-type=OSNATIVE in the latest dsu, the default just changed to repository.

khappone commented:
I think in locked down environments where you need a proxy, the problem comes from 
"Downloading the Index catalog"
I don't think this comes from repos regardless of source-type, so DSU itself needs to be given the proxy information.

khappone commented:
> I think in locked down environments where you need a proxy, the problem comes from "Downloading the Index catalog" I don't think this comes from repos regardless of source-type, so DSU itself needs to be given the proxy information.

You're right, I'm wrong. Using OSNATIVE does seem to skip that part

khappone committed:
Changed approach for the DSU to use source-type

chagned the approach to use source-type instead of setting proxies for
dsu runs. Still made a root alias so dsu can be run from the command
line like before.
